deepinterpolation/inferrence_collection.py

```diff
@@ -233,45 +313,164 @@ def run(self):
         chunk_size = [1]
         chunk_size.extend(self.indiv_shape[:-1])
 
+        output_dataset_name = "data"
+        raw_dataset_name = "raw"
+
         with h5py.File(self.output_file, "w") as file_handle:
-            dset_out = file_handle.create_dataset(
-                "data",
+            file_handle.create_dataset(
+                output_dataset_name,
                 shape=tuple(final_shape),
                 chunks=tuple(chunk_size),
                 dtype=self.output_datatype,
             )
 
-            # Initialize onset of output sample movie
-            local_start = first_sample
+            if self.save_raw:
+                file_handle.create_dataset(
+                    raw_dataset_name,
+                    shape=tuple(final_shape),
+                    chunks=tuple(chunk_size),
+                    dtype=self.output_datatype,
+                )
+
+        logger.info(f"Created empty HDF5 file {self.output_file}")
+        
+        self.model = _load_model(self.json_data)
```

Comment:
please initialize to `nONe` in constructor

Reply:
Could you explain the rationale?

Reply:
Instance variables should be initialized in the constructor. It is just good practice. All state should be defined in the constructor.